libs/partners/elasticsearch/langchain_elasticsearch/vectorstores.py

```diff
@@ -1284,3 +1356,15 @@ def SparseVectorRetrievalStrategy(
                     deployed to Elasticsearch.
         """
         return SparseRetrievalStrategy(model_id=model_id)
+
+    @staticmethod
+    def BM25RetrievalStrategy(
+        k1: float = 2.0, b: float = 0.75
```

Comment:
might consider setting the defaults on the __init__ function, so that `BM25RetrievalStrategy()` is simple to use as well

Reply:
I suggest making these two arguments `None` by default, and only include them in the mapping definition when the user provided custom values. That would allow the ES service to apply its own defaults for the general case.

Reply:
The direct reason for setting these default values is because the [`ElasticsearchBM25Retriever`](https://github.com/langchain-ai/langchain/blob/40f846e65da37a1c00d72da9ea64ebb0f295b016/libs/community/langchain_community/retrievers/elastic_search_bm25.py) utilizes them. However, I completely agree with @miguelgrinberg 's viewpoint that the default values should be set to `None`, allowing Elasticsearch's default values to be used unless explicitly specified by the user. This approach will ensure that the behavior of `ElasticsearchStore` aligns more consistently with Elasticsearch itself.